File: Estudos - repositorio/Ferramentas de Tecnologia/Java/Desafios - guanabara/desafio_4.java
import java.util.Scanner;

public class desafio_4 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Entrada do usuário
        System.out.print("Digite algo: ");
        String str = scanner.nextLine();

        // Verificações diretamente no método main
        System.out.println("String original: " + str);
        System.out.println("String em maiúsculas: " + str.toUpperCase());
        System.out.println("É número? " + str.matches("\\d+"));
        System.out.println("É letra? " + str.matches("[a-zA-Z]+"));
        System.out.println("É alfanumérico? " + str.matches("[a-zA-Z0-9]+"));
        System.out.println("É decimal? " + str.matches("\\d+(\\.\\d+)?"));
        System.out.println("Está em maiúsculas? " + str.equals(str.toUpperCase()));
        System.out.println("Está em minúsculas? " + str.equals(str.toLowerCase()));
        System.out.println("Contém apenas espaços em branco? " + str.trim().isEmpty());

        scanner.close();
    }
}

File: Estudos - repositorio/Ferramentas de Tecnologia/C++/Desafios Guanabara/desafio_4.cpp
#include <iostream>
#include <string>
#include <cctype>
#include <algorithm>
#include <regex>
#include <windows.h>

using namespace std;

int main() {
    SetConsoleOutputCP(CP_UTF8);
    
    // Entrada do usuário
    cout << "Digite algo: ";
    string str;
    getline(cin, str); // outra forma de ler entradas do usuario

    // Verificar se é número
    bool isNumber = regex_match(str, regex("\\d+"));

    // Verificar se é letra
    bool isAlphabet = regex_match(str, regex("[a-zA-Z]+"));

    // Verificar se é alfanumérico
    bool isAlphanumeric = regex_match(str, regex("[a-zA-Z0-9]+"));

    // Verificar se é decimal
    bool isDecimal = regex_match(str, regex("\\d+(\\.\\d+)?"));

    // Verificar se está em maiúsculas
    string upperStr = str;
    transform(upperStr.begin(), upperStr.end(), upperStr.begin(), ::toupper);
    bool isUpperCase = (str == upperStr);

    // Verificar se está em minúsculas
    string lowerStr = str;
    transform(lowerStr.begin(), lowerStr.end(), lowerStr.begin(), ::tolower);
    bool isLowerCase = (str == lowerStr);

    // Verificar se contém apenas espaços em branco
    bool isWhitespace = all_of(str.begin(), str.end(), ::isspace);

    // Resultados
    cout << "String original: " << str << endl;
    cout << "String em maiúsculas: " << upperStr << endl;
    cout << "É número? " << (isNumber ? "Sim" : "Não") << endl;
    cout << "É letra? " << (isAlphabet ? "Sim" : "Não") << endl;
    cout << "É alfanumérico? " << (isAlphanumeric ? "Sim" : "Não") << endl;
    cout << "É decimal? " << (isDecimal ? "Sim" : "Não") << endl;
    cout << "Está em maiúsculas? " << (isUpperCase ? "Sim" : "Não") << endl;
    cout << "Está em minúsculas? " << (isLowerCase ? "Sim" : "Não") << endl;
    cout << "Contém apenas espaços em branco? " << (isWhitespace ? "Sim" : "Não") << endl;

    return 0;
}
